oauth2_client/providers/salesforce/provider.py

```diff
@@ -0,0 +1,151 @@
+"""
+SalesforceProvider.
+Currently supports only JWT Token Bearer OAuth2 flow.
+"""
+import json
+from base64 import urlsafe_b64encode
+from datetime import datetime, timedelta
+
+import requests
+from Crypto.Hash import SHA256
+from Crypto.PublicKey import RSA
+from Crypto.Signature import PKCS1_v1_5
+
+from labsterutils.obj import get_value_or_default
+from oauth2_client.models import AccessToken
+from oauth2_client.providers.oauth2.provider import OAuth2Provider
+from oauth2_client.compat import timestamp
+
+
+class SalesforceProvider(OAuth2Provider):
+    id = 'salesforce'
+    name = 'Salesforce'
+
+    JWT_HEADER = '{"alg":"RS256"}'
+    GRANT_TYPE = "urn:ietf:params:oauth:grant-type:jwt-bearer"
+    AUDIENCE_URI = 'https://login.salesforce.com'
+    SANDBOX_AUDIENCE_URI = 'https://test.salesforce.com'
+    AUTH_URI = 'https://login.salesforce.com/services/oauth2/token'
+    SANDBOX_AUTH_URI = 'https://test.salesforce.com/services/oauth2/token'
```

Comment:
Aren't all these data should come from Application model?

Reply:
I answered in Confluence comment: https://liv-it.atlassian.net/wiki/spaces/WEB/pages/891191387/OAuth2Client+documentation?focusedCommentId=894238748#comment-894238748 

In summary: I think they can stay in the code in case of SF provider.

Reply:
Removed from the code. Marking this as resolved.